### PR TITLE
Remove deprecated SystemNavigator.routeUpdated method

### PR DESCRIPTION
**Pull request by goderbauer**
The deprecation period has elapsed. The method was made obsolete in https://github.com/flutter/flutter/pull/82594.

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
It looks like this one could also benefit from a dart fix to replace the old method with the new one

**Comment by goderbauer** [comment]
I doubt that this one is worth the effort. It's a fairly low-level method that was likely only used directly within the framework...

**Review by Piinks — COMMENTED** [review]
I agree, but this is the first batch of deprecations that we can have 100% dart fix support. I know it is trivial, but it is really nice to be able to say we have 100% coverage. 😁

File: packages/flutter/lib/fix_data/fix_services.yaml [hunk 1]
```diff
@@ -18,6 +18,31 @@
...
+      - kind: 'removeParameter'
+        name: 'previousRouteName'
+      - kind: 'removeParameter'
+        name: 'routeName'
+      - kind: 'addParameter'
+        index: 0
+        name: 'location'
+        style: optional_named
+        argumentValue:
+          expression: '{% location %}'
```
Comment (goderbauer) [inline packages/flutter/lib/fix_data/fix_services.yaml:38]: This part doesn't seem to work, the parameter doesn't get added.

Comment (Piinks) [reply packages/flutter/lib/fix_data/fix_services.yaml:38]: I think the `requiredIf:  key is necessary since it is an optional parameter

Comment (goderbauer) [reply packages/flutter/lib/fix_data/fix_services.yaml]: Tried that in https://github.com/flutter/flutter/pull/119187/commits/a18622d007170d030188e1b5171cde38c465c744, it didn't work :/

**Review by Piinks — COMMENTED** [review]
Filed https://github.com/dart-lang/sdk/issues/51198 for tracking, in the meantime this can land without the fix and we will follow up.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Pull request merged**